content/blog/getting-started-with-k8s-part4/index.md

```diff
@@ -0,0 +1,95 @@
+
+- [Building a Kubernetes cluster on cloud providers]({{< relref "/blog/getting-started-with-k8s-part1" >}})
+- [Basic application deployment]({{< relref "/blog/getting-started-with-k8s-part2" >}})
+- [Advance application deployment and Helm charts]({{< relref "/blog/getting-started-with-k8s-part3" >}})
+- Stateful applications
+- Networking
+- “Day 2” activities such as migrating node groups.
+
+In the previous post, we examined different methods for deploying applications. We worked through examples of a boilerplate deployment, to one using `ComponentResources` to automate deployment further, and deploying with Helm charts. In this installment, we’ll look at how to deploy stateful applications, such as databases, in Kubernetes. Unlike stateless applications, stateful apps require persistent and durable resources such as storage, which presents scaling and availability challenges.
+
+<!--more-->
+
+## Stateful vs. Stateless
+
+Stateless applications don’t require information from a previous transaction because they are single request and response. For example, the [“Hello World” example](https://www.pulumi.com/blog/getting-started-with-k8s-part2/#examples) that we deployed in an earlier article is a stateless application. The critical thing to remember is that in Kubernetes, any container with that application in any pod can respond to that request.
+Stateful applications can read or store information about previous transactions, and a transaction is executed within the context of a previous extraction. Email is an example of a stateful application, e.g., an email can be unread or read, or it can be grouped according to keywords or by content. Unlike stateless applications in Kubernetes, the requests are sent to the same Pod.
+
+## Stateful sets
+
+Stateful sets are the Kubernetes workload object for managing stateful applications. They are an instance of [Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), where Pods have an identical container spec but maintain a sticky or persistent identity for each Pod maintained across rescheduling when a Pod fails or is terminated.
+
+You would use a [StatefulSet for](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#using-statefulsets):
+stable and unique network identifiers
+stable and persistent storage
+applications that require ordered deployment and scaling
+applications that require ordered and rolling updates
+
+When a StatefulSet is created, each pod is assigned a unique and persistent identifier. Pods are named numerically starting from 0, e.g., pod-0, pod-1, pod-n. StatefulSets use a [Headless Service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) resource, which sets the `clusterIP` property ton `none`. Pods in a StatefulSet will have an IP address mapped directly to instead of a virtual IP address. This lets other Pods in the cluster directly communicate to an application, as in database connections.
```

Comment:
```suggestion
When a StatefulSet is created, each Pod is assigned a unique and persistent identifier. Pods are named numerically starting from 0, e.g., pod-0, pod-1, pod-n. StatefulSets use a [Headless Service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) resource, which sets the `clusterIP` property to `none`. Pods in a StatefulSet will have an IP address mapped directly to them instead of a virtual IP address. This lets other Pods in the cluster directly communicate to an application, as in database connections.
```